p2p/transport/webrtc/errors.go

```diff
@@ -0,0 +1,49 @@
+package libp2pwebrtc
+
+import (
+	"fmt"
+)
+
+type errKind string
+
+const (
+	errKindConnectionFailed errKind = "peerconnection failed"
+	errKindDatachannel      errKind = "datachannel"
+	errKindMultiaddr        errKind = "bad multiaddr"
+	errKindNoise            errKind = "noise"
+	errKindInternal         errKind = "internal"
+)
+
+var (
+	errDataChannelTimeout = errDatachannel("timed out waiting for datachannel", nil)
+)
+
+type webRTCTransportError struct {
+	kind    errKind
+	message string
+	nested  error
+}
+
+func (e *webRTCTransportError) Error() string {
```

Comment:
Why is that taking a pointer receiver ? You are never modifying any of the fields.

Reply:
@Jorropo these will go away in favor of `fmt.Errorf` calls as per @marten-seemann 's review.